### ISSUE TITLE
Roadmap?

### ISSUE THREAD

**Issue by voltechs** [jiahaog/nativefier#1067]
**Homework**

- [x] I looked at `nativefier --help` and https://github.com/jiahaog/nativefier/blob/master/docs/api.md
- [x] I searched existing issues, open & closed. Yes, my question is new.
- [x] I'm using the latest version available at https://github.com/jiahaog/nativefier/releases

**Your question**

(Following meant as compassionate/constructive questions/rhetoric)
Is there a roadmap for the vision of `nativefier`? Additionally, and more importantly, are there any/better ways for the community to contribute? I've noticed a load of opened issues that haven't really been triaged or addressed. Currently 5 outstanding PRs (far less than some other projects, but we're not comparing here)

I'm personally really excited about this project. The two biggest wish-lists for me are "global/universal" runtime (reduce the size of these apps), and better support for injections (I still haven't been able to get JS/CSS injection working, and solutions/information seems scattered and scarce) including chrome extensions such as adblock etc.

I do see some activity on the project, lately largely around refactoring and modernizing/updating dependencies. This is great, but it's also not really pushing the project/product forward very much (IMO). Believe me, though, I'm very passionate about updating outdated dependencies.

**Steps to reproduce**

Peruse [Issues](https://github.com/jiahaog/nativefier/issues) and [Pull Requests](https://github.com/jiahaog/nativefier/pulls).

**TL;DR**

Is there a roadmap that can be shared with the community? Is there room for assistance from the community in triaging issues, giving feedback back to Pull Requests and getting them ready for merging? All of this of course would be better served with a roadmap/general sense of direction for where the project is going.

Once again as a disclaimer, I'm very excited about this project and I love where it is, and where it's come from. _My_ perception is that it's stagnated a bit, and my motivation is to open up a conversation about moving it forward, or at least better understanding what the plans are for the project and how I/the community can help!

Thanks so much for the creation and maintenance of the project thus far!

❤️

**Comment by ronjouch** [comment]
@voltechs see [blurb in recent releases](https://github.com/jiahaog/nativefier/releases), it was refined to express concisely the current state of Nativefier:

> 👋 And a reminder: dear users / contributors! Public service announcement!
> 
> Nativefier is maintained by YOU. It's reasonably sane to hack on since
> the 8.x cleanup, it fits the use cases needed by the original author and
> the current maintainer, so _we_ are not doing any active development.
> 
> But it's not dead! If YOU use it and want to see juicy features & fixes,
> PR welcome! In particular, here are three particularly annoying issues;
> help welcome on those (or any other issue that motivates you 🙂):
> 
> - https://github.com/jiahaog/nativefier/issues/902
> - https://github.com/jiahaog/nativefier/issues/831
> - https://github.com/jiahaog/nativefier/issues/706
> - For other issues, https://github.com/jiahaog/nativefier/issues
> 
> Thanks! Take care.

---

And here's a point-by-point answer:

> (Following meant as compassionate/constructive questions/rhetoric)

Thaaaaaaaaaaanks, and great job on communicating your question, it's 100% clear you're compassionate, constructive, and taking your time to make your point 🙂.

> Is there a roadmap for the vision of nativefier?

Not from me, and not from the original author (Jia). These days I barely use it (thanks, Firefox Containers!), so my level of interest is close to zero.

> Additionally, and more importantly, are there any/better ways for the community to contribute? I've noticed a load of opened issues that haven't really been triaged or addressed. Currently 5 outstanding PRs (far less than some other projects, but we're not comparing here)

- **PRs**: the amount of incoming PRs is low enough for me to handle currently, so A. I'm not looking for help, but B. if you want to "finish" one or two of the currently-open PRs, go for it: fork the branch, address the issues, and open a new PR 🙂. For context, all 5 open PRs are left unmerged because they're unsatisfying. Sometimes they're marked as "changes requested", sometimes it's just in the comments, but see for yourself: none is mergeable. I should close them (since they're so old) with a nice _"Hey sorry, closing to clean up, will re-open if you address the issues"_ message like I did a few times in the past; I'm irrational about it because I don't like to reject / piss off PR authors :shrug: .
- **Issues**: here I could take some help, yes 🙂. If you have the bandwidth, consider watching the repo (top button `Watch` → `Watching: Be notified of all conversations`) and responding to new issues. After I've seen you help on a few issues, I'll be happy to ask Jia to give you issue triaging rights.

> I'm personally really excited about this project. The two biggest wish-lists for me are "global/universal" runtime (reduce the size of these apps), and better support for injections (I still haven't been able to get JS/CSS injection working, and solutions/information seems scattered and scarce) including chrome extensions such as adblock etc.

All issues I'd love to see addressed! Go for them and open PRs! Then, same for issues above: be a Crazy Patch Writer™, get a few PRs merged, then when you have established you understand the project, ask Jia for a commit bit 🙂.

> I do see some activity on the project, lately largely around refactoring and modernizing/updating dependencies. This is great, but it's also not really pushing the project/product forward very much (IMO). Believe me, though, I'm very passionate about updating outdated dependencies.

The reasoning behind the 8.0 TS refactor and regular Electron/deps upgrades is the same: I'm not helping "raise" Nativefier (because, again, I barely use it), but I at least want to keep it healthy and hackable. So, this is my current "minimal maintainer" budget:

1. Stay afloat major dependencies bumps and Electron breaking changes.
2. Do occasional issue triaging.
3. Merge mergeable PRs.

> Thanks so much for the creation and maintenance of the project thus far!

Thank you! Hope this clarifies where it is now 🙂.

**Comment by ronjouch** [comment]
@voltechs with that, closing, but I get notifications and will read your feedback if you comment.

**Comment by voltechs** [comment]
@ronjouch thanks! I appreciate the comments. Been slammed at work (much to my own chagrin), but I will definitely revisit!
